pkg/planner/cascades/memo/group_expr.go

```diff
@@ -28,40 +27,36 @@ import (
 // coupling between Group and GroupExpression is the key to the success of the memory compact
 // of representing a forest.
 type GroupExpression struct {
+	// LogicalPlan is internal logical expression stands for this groupExpr.
+	// Define it in the header element can make GE as Logical Plan implementor.
+	base.LogicalPlan
+
 	// group is the Group that this GroupExpression belongs to.
 	group *Group
 
 	// inputs stores the Groups that this GroupExpression based on.
 	Inputs []*Group
 
-	// logicalPlan is internal logical expression stands for this groupExpr.
-	logicalPlan base.LogicalPlan
-
 	// hash64 is the unique fingerprint of the GroupExpression.
 	hash64 uint64
 }
 
-// GetLogicalPlan returns the logical plan of the GroupExpression.
-func (e *GroupExpression) GetLogicalPlan() base.LogicalPlan {
-	return e.logicalPlan
-}
-
 // GetGroup returns the Group that this GroupExpression belongs to.
 func (e *GroupExpression) GetGroup() *Group {
 	return e.group
 }
 
 // String implements the fmt.Stringer interface.
-func (e *GroupExpression) String(w io.Writer) {
-	e.GetLogicalPlan().ExplainID()
-	_, _ = w.Write([]byte("GE:" + e.GetLogicalPlan().ExplainID().String() + "{"))
+func (e *GroupExpression) String(w util.IBufStrWriter) {
+	e.LogicalPlan.ExplainID()
+	w.WriteString("GE:" + e.LogicalPlan.ExplainID().String() + "{")
 	for i, input := range e.Inputs {
 		if i != 0 {
-			_, _ = w.Write([]byte(", "))
+			w.WriteString(", ")
 		}
 		input.String(w)
 	}
-	_, _ = w.Write([]byte("}"))
+	w.WriteString("}")
 }
 
 // Sum64 returns the cached hash64 of the GroupExpression.
```

Comment:
What the different mean between sum64 and hash64 ?

Reply:
the former one is like a GET utility, the later is compute function.